src/components/tweet-card/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import tweet from "../../__mock__/tweet.json";
+
+import {
+  Card,
+  AvatarContainer,
+  Tweet,
+  TweetContainer,
+  UserInfo,
+  InfoConatiner,
+  CreatedAt,
+  TextContainer,
+  FeedbackContainer,
+  Feedback,
+  Followers,
+  FollowersContainer,
+} from "./style";
+import {
+  HeartIcon,
+  MoreIcon,
+  ReplyIcon,
+  RetweetIcon,
+  ShareIcon,
+} from "../../icons/styled-icons";
+import UserAvatar from "../common/avatar";
+import Title from "../common/Title";
+import UserName from "../common/user-name";
+import moment from "moment";
+
+const TweetCard: React.FC = () => {
+  const [createdAt, setCreatedAt] = useState("");
+  const clickHandler = (e: any) => {
+    const el = e.target.closest("span");
+    if (el && e.currentTarget.contains(el)) {
+      console.log(el.textContent);
+    }
+  };
+  const getCreatedAt = () => {
+    const duration = moment.duration(
+      moment(new Date()).diff(moment(tweet.data.created_at))
+    );
+    const hours = +duration.asHours();
+    if (hours < 0) {
+      setCreatedAt(String(hours % 60));
+    } else if (hours > 24) {
+      setCreatedAt(moment(tweet.data.created_at).format("MMMM DD"));
+    } else setCreatedAt(String(hours));
+  };
```

Comment:
this method is transforming date into a particular format, please make it a pure function rather than using state.